### PR TITLE
Add & Delete command Abstractions

### LINKED ISSUE #277: Update Junjie's PPP
Portfolio project page

### LINKED ISSUE #281: Abstract add command and delete command as abstract classes

### PR DESCRIPTION
**Pull request by Asuraxsoul**
Abstracted both AddCommand and DeleteCommand classes to be abstract class and any sub commands under them previously were abstracted out as a Command of their own.
Fixes #281 
Update JunJie's PPP.
Fixes #277

### CODE REVIEW

File: docs/team/asuraxsoul.md [hunk 1]
```diff
@@ -5,26 +5,46 @@ title: Lim Jun Jie's Project Portfolio Page
...
+  * Highlights: Similar to `Add Task` feature, this feature allows me to understand the main architecture of JARVIS.
+
+* **New Feature**: Added `View Task` feature for Tutors to view their different subsets of their personal tasks.
+  * What it does: Allows tutors to view different Task list including `Overall Task List`, `Todo List`, `Event List`
+  and `Deadline List`.
+  * Justification: This feature gives tutors ability to view their Tasks quickly based on different categories.
+  * Highlights: I really appreciate the usefulness and flexibility of using an Observer pattern.
+
+* **New Feature**: Added a `history command` that allows Tutors to navigate to previous commands using up/down keys.
+  * What it does: Allows tutors to navigate through previous commands entered using up and down arrow keys.
```
Comment (peironggg) [inline docs/team/asuraxsoul.md:32]: Great detail for you PPP!

File: src/main/java/seedu/jarvis/logic/commands/add/AddConsultationCommand.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package seedu.jarvis.logic.commands.add;
```
Comment (ngzhenteng) [inline src/main/java/seedu/jarvis/logic/commands/add/AddConsultationCommand.java:1]: Great use of Abstractions

Comment (peironggg) [inline src/main/java/seedu/jarvis/logic/commands/add/AddConsultationCommand.java:1]: Great abstraction taken to separate out the Add commands 👍

File: src/main/java/seedu/jarvis/logic/commands/delete/DeleteConsultationCommand.java [hunk 3]
```diff
@@ -1,12 +1,26 @@
 package seedu.jarvis.logic.commands.delete;
```
Comment (peironggg) [inline src/main/java/seedu/jarvis/logic/commands/delete/DeleteConsultationCommand.java:1]: Great work to abstract away the Delete commands 🥇 It is definitely not an easy task.

File: src/main/java/seedu/jarvis/logic/commands/add/AddTaskCommand.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
+package seedu.jarvis.logic.commands.add;
```
Comment (ngzhenteng) [inline src/main/java/seedu/jarvis/logic/commands/add/AddTaskCommand.java:1]: Another great use of abstractions, this makes the code neater. thank you!

**Review by ngzhenteng — APPROVED** [review]
LGTM, thank you!

File: src/test/java/seedu/jarvis/logic/commands/AddCommandIntegrationTest.java [hunk 5]
```diff
@@ -1,15 +1,11 @@
 package seedu.jarvis.logic.commands;
 
-import static seedu.jarvis.logic.commands.CommandTestUtil.assertCommandFailure;
```
Comment (peironggg) [inline src/test/java/seedu/jarvis/logic/commands/AddCommandIntegrationTest.java:3]: Thanks for adapting the tests to fit the new Command structure.

**Comment by Asuraxsoul** [comment]
THANK you all!

**Pull request merged**